google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageOptions.java

```diff
@@ -207,13 +208,23 @@ private Tuple<StorageSettings, Opts<UserProject>> resolveSettingsAndOpts() throw
       }
     }
 
+    boolean isTm =
+        Arrays.stream(Thread.currentThread().getStackTrace())
+            .anyMatch(
+                ste -> ste.getClassName().startsWith("com.google.cloud.storage.transfermanager"));
+
     HeaderProvider internalHeaderProvider =
         StorageSettings.defaultApiClientHeaderProviderBuilder()
             .setClientLibToken(ServiceOptions.getGoogApiClientLibName(), getLibraryVersion())
             .build();
+    if (isTm) {
+      internalHeaderProvider =
+          XGoogApiClientHeaderProvider.of(
+              internalHeaderProvider, ImmutableList.of("gccl-gcs-cmd/tm"));
```

Comment:
There's 4 different TM header options:
```
tm.upload_many
tm.download_many
tm.upload_sharded
tm.download_sharded
```
Are you planning to support this? cc: @andrewsg

Reply:
The lifecycle of this header in java is only as granular as the client instance. We do not have the means of doing per invocation selection. This limitation was mentioned back during the design phase, and is part of why there is `tm.` to allow prefix course selection while allowing fine grained selection for those languages where it is possible.